app/src/main/java/com/example/pygmyhippo/organizer/EventFragment.java

```diff
@@ -333,26 +335,60 @@ public void drawWinners(Event event) {
         Random rand = new Random();
         int winnerNumber = event.getCurrentWinners();
 
-        // Loop for how many winners this event wants (using winner count to keep check on it)
-        while (winnerNumber < event.getEventWinnersCount()) {
-            // Only try to draw applicants if there are entrants in the waitlist
-            if (event.getNumberWaitlisted() > 0) {
-                // Draw a random index number between 0 and the size of the list
-                int drawIndex = rand.nextInt(event.getEntrants().size());
-
-                if (!event.getEntrants().get(drawIndex).getEntrantStatus().value.equals("waitlisted")) {
-                    // This entrant does not have the status for a draw/redraw, so skip them
-                    continue;
+        if (event.getEventStatus().value.equals("ongoing")) {
+            // For initial draw
+            while (winnerNumber < event.getEventWinnersCount()) {
+                // Only try to draw applicants if there are entrants in the waitlist
+                if (event.getNumberWaitlisted() > 0) {
+                    // Draw a random index number between 0 and the size of the list
+                    int drawIndex = rand.nextInt(event.getEntrants().size());
```

Comment:
are event entrants initialized to an empty array or null, i could see a null exeption could happen here if entrant list was null